Enhancement: Specify error message for reschedule command (#302)
AY2324S2-CS2103-F15-1/tp: docs/DeveloperGuide.md

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

### 5. Specify error message for reschedule and schedule commands

**Current Implementation**:

> - When users input a invalid date for any parameters, the error message states that it is an invalid command format, although the issue is that an invalid date is passed.

Reviewer: ```suggestion
- When users input an invalid `START_DATETIME` or `END_DATETIME`, an invalid command format error message is shown instead of an invalid datetime error message.
- The error message is not representative of the error and should be more specific about which fields are incorrect.
```
Maybe this is a better phrasing?

Author: Clearer!

**Proposed Implementation**:
- The error message should specify which of the given parameters are failing instead of prompting an invalid command format.

> - This can be applied for both reschedule and schedule as they go through the same checks.

Reviewer: I think the phrasing of the proposed implementation can be improved? @Dethada @Javiery3889 @Arixeyeion Do you have any suggestions for improvement?

Reviewer 2: For line 441
```suggestion
- The error message should mention which parameter failed and the reason for the failure, e.g. `The DateTime 31-02-2024T12:00 is invalid for START_DATETIME parameter. Please use the following format: dd-MM-yyyy'T'HH:mm, e.g. 02-02-2024T22:00.`
```

Reviewer: I think the phrasing is good, but example can be improved? Perhaps a shorter message would be better such as `START_DATETIME is invalid`?

Do we need multiple error messages, where one is for incorrect `DATETIME` format and another is for incorrect `DATETIME` value about a particular `DATETIME`?

Author: I think better to have both.

Reviewer 2: @KohGuanZeh @YuZizhen 
I would suggest:
- The error message should mention which parameter failed and the reason for the failure, e.g. `The DateTime 31-02-2024T12:00 is invalid. Please use the following format: dd-MM-yyyy'T'HH:mm, e.g. 02-02-2024T22:00.`

Because the command is left in the Commnd Box when an error occurs, so it is sufficient for the user input to be displayed back to the user for the DateTime that cannot be parsed.

- Modify the format of `DATE` to be `d-M-yyyy` and `DATETIME` to be `d-M-yyyy`T`H:mm` instead.
- This allows FINDvisor to accept both single and double-digits day, month and hour values as valid `DATE` and `DATETIME` values and would not require users to pad these single digit values with a leading zero.

> ### 5. Specify error message for reschedule and schedule commands

Reviewer 2: ```suggestion
### 5. Specify error message for parsing DateTime strings in `schedule` and `reschedule` commands
```
